src/vmm/src/builder.rs

```diff
@@ -696,14 +681,16 @@ fn attach_block_devices(
             .open(&drive_config.path_on_host)
             .map_err(OpenBlockDevice)?;
 
-        if drive_config.is_root_device && drive_config.get_partuuid().is_some() {
+        if drive_config.is_root_device {
             let kernel_cmdline = &mut vmm.kernel_cmdline;
 
-            kernel_cmdline.insert_str(format!(
-                "root=PARTUUID={}",
+            kernel_cmdline.insert_str(if drive_config.get_partuuid().is_some() {
                 //The unwrap is safe as we are firstly checking that partuuid is_some().
-                drive_config.get_partuuid().unwrap()
-            ))?;
+                format!("root=PARTUUID={}", drive_config.get_partuuid().unwrap())
+            } else {
+                // If no PARTUUID was specified for the root device, try with the /dev/vda.
+                "root=/dev/vda".to_string()
+            })?;
```

Comment:
To avoid that `unwrap()`, you can replace these lines with:

`
kernel_cmdline.insert_str(if let Some(partuuid) = drive_config.get_partuuid() {
                format!("root=PARTUUID={}", partuuid)
            } else {
                // If no PARTUUID was specified for the root device, try with the /dev/vda.
                "root=/dev/vda".to_string()
            })?;
`